### PR TITLE
Fix path for hip-clang when using hipcc

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

File: hip-config-clang.cmake.in [hunk 1]
```diff
@@ -50,14 +50,24 @@ set_and_check( hip_BIN_INSTALL_DIR "@PACKAGE_BIN_INSTALL_DIR@" )
 set_and_check(hip_HIPCC_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipcc")
 set_and_check(hip_HIPCONFIG_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipconfig")
 
-get_filename_component(HIP_CLANG_ROOT "${CMAKE_CXX_COMPILER}" PATH)
-get_filename_component(HIP_CLANG_ROOT "${HIP_CLANG_ROOT}" PATH)
+if(CMAKE_CXX_COMPILER MATCHES ".*hipcc")
+  execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version 
+                  OUTPUT_STRIP_TRAILING_WHITESPACE
+                  OUTPUT_VARIABLE HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT)
+  if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHER "InstalledDir: ([^\t\r\n])")
```
Comment (yxsamliu) [inline hip-config-clang.cmake.in]: I tried this and got error:
/opt/rocm/hip/lib/cmake/hip/hip-config.cmake(78):  execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version OUTPUT_STRIP_TRAILING_WHITESPACE OUTPUT_VARIABLE HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT )
/opt/rocm/hip/lib/cmake/hip/hip-config.cmake(81):  if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHER InstalledDir: ([^\t\r\n]) )
CMake Error at /opt/rocm/hip/lib/cmake/hip/hip-config.cmake:81 (if):
  if given arguments:

    "HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT" "MATCHER" "InstalledDir: ([^	

  ])"

  Unknown arguments specified
Call Stack (most recent call first):
  cmake/VerifyCompiler.cmake:58 (find_package)
  CMakeLists.txt:47 (include)


-- Configuring incomplete, errors occurred!

Comment (yxsamliu) [reply hip-config-clang.cmake.in]: there is a typo:
MATCHER -> MATCHES

Comment (pfultz2) [reply hip-config-clang.cmake.in]: Oops that should be `MATCHES`

File: hip-config-clang.cmake.in [hunk 2]
```diff
@@ -50,14 +50,24 @@ set_and_check( hip_BIN_INSTALL_DIR "@PACKAGE_BIN_INSTALL_DIR@" )
 set_and_check(hip_HIPCC_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipcc")
 set_and_check(hip_HIPCONFIG_EXECUTABLE "${hip_BIN_INSTALL_DIR}/hipconfig")
 
-get_filename_component(HIP_CLANG_ROOT "${CMAKE_CXX_COMPILER}" PATH)
-get_filename_component(HIP_CLANG_ROOT "${HIP_CLANG_ROOT}" PATH)
+if(CMAKE_CXX_COMPILER MATCHES ".*hipcc")
+  execute_process(COMMAND ${CMAKE_CXX_COMPILER} --version 
+                  OUTPUT_STRIP_TRAILING_WHITESPACE
+                  OUTPUT_VARIABLE HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT)
+  if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHES "InstalledDir: ([^\t\r\n])")
```
Comment (yxsamliu) [inline hip-config-clang.cmake.in]: need this:

if(HIP_CLANG_CXX_COMPILER_VERSION_OUTPUT MATCHES "InstalledDir:[\t\r\n][\t\r\n]*([^\t\r\n])")

verified with this change it works.

**Pull request merged**